pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -80,7 +80,8 @@ func Add(mgr manager.Manager, opts manager.Options) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager, opts manager.Options) (*ReconcileMachineHealthCheck, error) {
-	if err := mgr.GetCache().IndexField(&mapiv1.Machine{},
+	if err := mgr.GetCache().IndexField(context.Background(),
```

Comment:
I would consider putting this as a TODO so we remember to fix it up at some point
```suggestion
	if err := mgr.GetCache().IndexField(context.TODO(),
```